src/SchemaExplorer.tsx

```diff
@@ -447,61 +439,79 @@ export class SchemaExplorer extends React.PureComponent<SchemaExplorerProps, Sch
     max-width: 100%;
   `;
 
-  private static HeadingContainer = styled.div`
+  public static HeadingContainer = styled.div`
     display: flex;
     flex-direction: row;
     justify-content: space-between;
   `;
 
-  private static Heading = styled.h1`
+  public static Heading = styled.h1`
     font-size: 16px;
     font-weight: 600;
     padding-top: 24px;
     margin: 5px 8px;
   `;
 
-  UNSAFE_componentWillMount() {
-    this.setState({
+  constructor(props: SchemaExplorerProps) {
+    super(props);
+    this.state = {
       pathExpanded: false,
       view: 'details'
-    });
+    }
   }
 
   render() {
-    const { path, schema, lookup, stage, basePathSegments } = this.props;
+    const { path, schema, lookup, stage, basePathSegments, validationResults, onSelectValidationRange } = this.props;
     const { pathExpanded } = this.state;
     if (path.length === 0) {
       return <div>TODO What do we do when the reference could not be found? Error maybe?</div>;
     }
 
     const currentPathElement = path[path.length - 1];
 
-    const tabData: TabData[] = [{
-      label: 'Details',
-      content: (
-        <SchemaExplorerDetails
-          schema={schema}
-          reference={currentPathElement.reference}
-          lookup={lookup}
-          stage={stage}
-          clickElement={createClickElement({ basePathSegments, path })}
-        />
-      )
-    }, {
-      label: 'Example (JSON)',
-      content: (
-        <SchemaExplorerExample schema={schema} lookup={lookup} stage={stage} format="json" />
-      )
-    }, {
-      label: 'Example (YAML)',
-      content: (
-        <SchemaExplorerExample schema={schema} lookup={lookup} stage={stage} format="yaml" />
-      )
-    }];
-
-    const onTabSelect: OnSelectCallback = tab => {
+    type ExtendedTabData = TabData & {
+      view: ViewType;
```

Comment:
Clever. That is one way of doing this without the clunky lookup that I have above.